crates/pallet-transaction-fees/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright (C) 2021 Subspace Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Pallet for charging and re-distributing transaction fees.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![forbid(unsafe_code)]
+#![warn(rust_2018_idioms, missing_debug_implementations)]
+
+mod default_weights;
+
+use frame_support::sp_runtime::traits::Zero;
+use frame_support::traits::{Currency, Get};
+use frame_support::weights::Weight;
+pub use pallet::*;
+use sp_consensus_subspace::digests::CompatibleDigestItem;
+
+type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+pub trait WeightInfo {
+    fn on_initialize() -> Weight;
+}
+
+#[frame_support::pallet]
+mod pallet {
+    use super::{BalanceOf, WeightInfo};
+    use frame_support::pallet_prelude::*;
+    use frame_support::traits::Currency;
+    use frame_system::pallet_prelude::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// `pallet-transaction-fees` events
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Minimum desired number of replicas of the blockchain to be stored by the network,
+        /// impacts storage fees.
+        #[pallet::constant]
+        type MinReplicationFactor: Get<u16>;
+
+        /// How much (ratio) of storage fees escrow should be given to farmer each block as a
+        /// reward.
+        #[pallet::constant]
+        type StorageFeesEscrowBlockReward: Get<(u64, u64)>;
+
+        /// How much (ratio) of storage fees collected in a block should be put into storage fees
+        /// escrow (with remaining issued to farmer immediately).
+        #[pallet::constant]
+        type StorageFeesEscrowBlockTax: Get<(u64, u64)>;
+
+        /// How many credits there is in circulation.
+        #[pallet::constant]
+        type CreditSupply: Get<BalanceOf<Self>>;
```

Comment:
I know it makes no difference to the compiler, but using a dedicated trait instead of misusing `Get` can achieve better readability IMHO.

Reply:
This is a canonical way in Substrate to use `Get` trait for constants. What would be the value of changing it?

Reply:
As you said it's typically for constants, but it's not the case here, so I prefer to make it more explicit and possibly reduce some subtle misunderstanding. It's just for human not for machine.

Reply:
Why is it not the case here? This is a constant and `Get<BalanceOf<Self>>` is fairly readable to me.

Reply:
No, the total issuance is definitely not a constant.

Reply:
I agree, but at the same time these constants make it possible for frontend to render things like total space pledged and cost of storage very easily and it is the way Substrate already works, so not really anything new.